tests/e2e/common/setup.rs

```diff
@@ -0,0 +1,88 @@
+use std::sync::Arc;
+
+use deadpool_diesel::{postgres::Pool, Manager};
+use testcontainers::ContainerAsync;
+use testcontainers_modules::kafka::Kafka;
+use testcontainers_modules::zookeeper::Zookeeper;
+
+use crate::common::containers::{
+    kafka::setup_kafka,
+    offchain_db::setup_offchain_db,
+    onchain_db::{run_onchain_migrations, setup_onchain_db},
+    pragma_node::{setup_pragma_node, PragmaNode},
+    zookeeper::setup_zookeeper,
+    Containers, Timescale,
+};
+use crate::common::logs::init_logging;
+
+/// Main structure that we carry around for our tests.
+/// Contains some usefull fields & functions attached to make testing easier.
+pub struct TestHelper {
+    pub node_base_url: String,
+    pub onchain_pool: Pool,
```

Comment:
shouldn't that be in an Arc<> ? If we are running our tests in parallel

Reply:
I don't think we want to do that - I can wrap them into Arc but ideally each test should be perfectly isoled & we should not use the pools in parallel

Reply:
but when we run tests in N threads do we have N TestHelper then ?

Reply:
Yup exactly, though it's not already working I think because we always assign the port `3000` and `8080` to `pragma_node` - if we were to run this async it would crash here:
```rust
        .with_mapped_port(SERVER_PORT, SERVER_PORT.tcp())
        .with_mapped_port(METRICS_PORT, METRICS_PORT.tcp())
```

But I can probably do like the databases & assign an available port

Reply:
I think it makes sense to just have a singleton here that share its members between threads in a safe way

Reply:
I think we should see when we have multiples tests if it make sense to run them async - I'm not sure it's worth it since it's gonna be fast anyway + less troubleshooting / debugging / weird async stuff when we're gonna deal with the databases (add data, assertions etc..)
wdyt?

Reply:
This could work as a workaround
https://github.com/la10736/rstest/issues/141#issuecomment-1372784171